lib/compiler/test/beam_validator_SUITE.erl

```diff
@@ -808,6 +810,17 @@ mrt_1(Bool) ->
     true = is_boolean(Bool),
     Bool.
 
+%% ERL-1340: bs_get_binary2 didn't match the SSA pass when updating match
+%% context units on tail matches ({atom,all}).
+bs_get_binary_context_type(Config) when is_list(Config) ->
+    [<<0,1,2,3,4>>, <<5,6,7,8,9>>] = bgbct_1(id(<<0,1,2,3,4,5,6,7,8,9>>)),
+    [] = bgbct_1(id(<<>>)),
+
+    ok.
+
+bgbct_1(<<Bin/binary>> = Bin) ->
```

Comment:
Shouldn't something like this be rejected by the linter in the first place, though?

Reply:
I suppose we could warn about it from OTP 24 onwards. As iffy as this seems it's still legal, and I'd rather not change it in a patch.

Reply:
I'm original reporter of this issue for Elixir (https://github.com/elixir-lang/elixir/issues/10309) and I want explain why I happened to write code like that.

I need this iffy expression to include argument name generated by ex_doc.
If I only include `<<bin::binary>>`, then argument in document is named as `arg`.
And I prefer `<<bin::binary>>` pattern matching to `is_binary(bin)` guard.

Reply:
@jechol Would it work for the documentation if you would write the function head like this?

    def encode_bin(<<_::binary>> = bin) do

The compiler will generate efficient code for that.

Reply:
@bjorng That's what I want. Thanks.

Reply:
I was specifically referring to variables on the left & right of patterns being "compared". I don't think it can ever succeed beyond this special case in binary where the pattern match is a no-op. Everywhere else a part of a data structure can't be equal to the data structure itself, e.g. `{X, _} = X` shouldn't even compile as it will always fail.

Reply:
@michalmuskala I think that `{X, _} = X` should produce a warning. That is what we do for other unmatchable patterns such as `[X,Y] = {X,Y}`. Our reasoning is that macros, parse transformations and naive code generators could produce nonsensical but working code.

Reply:
Introducing a warning in such cases would be a good improvement, agreed.